app/models/school.rb

```diff
@@ -878,6 +878,89 @@ def pupil_numbers_ok?
     end
   end
 
+  def self.from_onboarding(onboarding)
```

Comment:
You can initialize an object from a hash, and latter add additional attributes using `assign_attributes`, so would be clearer to write something like:

```
school = School.new({
 data_enabled: false,
 ...
})

return school if school.establishment.nil?

school.assign_attributes({
 establishment: ...,
 address: address_from_establishment(school.establishment)
 ...
})

```